molo/core/models.py

```diff
@@ -12,9 +14,31 @@
 from wagtail.wagtailimages.blocks import ImageChooserBlock
 
 from molo.core.blocks import MarkDownBlock
-
-
-class HomePage(Page):
+from molo.core import constants
+
+
+class CommentedPageMixin(object):
+    def get_effective_commenting_settings(self):
+        # return commenting settings for the homepage
+        if self.commenting_state:
+            return {
+                'state': self.commenting_state,
+                'open_time': self.commenting_open_time,
+                'close_time': self.commenting_close_time
+            }
+        # use the commenting settings for the parent page
+        parent_page = Page.objects.all().ancestor_of(self).last().specific
+        if (parent_page):
+            return parent_page.get_effective_commenting_settings()
+        # add a default in case nothing is set. We should never get here
+        return {
+            'state': constants.COMMENTING_DISABLED,
+            'open_time': None,
+            'close_time': None
+        }
+
+
+class HomePage(CommentedPageMixin, Page):
```

Comment:
The order is important here. I think you want them the other way around as the `Page` here is the main thing you want inherited and the Mixin only implements `get_effective_commenting_settings`. Python's MRO is super confusing and I don't fully get all of it but I think you want these as `(Page, CommentedPageMixin)` for further `super()` calls to work as expected.

I think @hodgestar knows more about the MRO, he did a CTPUG talk about it a while ago. I do suspect he has developed a dislike for Mixins as a result now though. YMMV :)